packages/lesswrong/components/review/ReviewVotingCanvas.tsx

```diff
@@ -0,0 +1,440 @@
+
+  const normalizeCoords = useCallback((clientX: number, clientY: number) => {
+    if (headerRef.current) {
+      const bounds = headerRef.current.getBoundingClientRect();
+      if (
+        clientX > bounds.left &&
+        clientX < bounds.right &&
+        clientY > bounds.top &&
+        clientY < bounds.bottom
+      ) {
+        return {
+          x: (clientX - bounds.left) / bounds.width,
+          y: (clientY - bounds.top) / bounds.height,
+        };
+      }
+    }
+    return null;
+  }, [headerRef]);
+
+  const addHeart = useCallback(async (x: number, y: number, theta: number) => {
+    const result = await rawAddHeart({
+      variables: {
+        electionName: "reviewVoting2022",
+        x,
+        y,
+        theta,
+      },
+    });
+    void refetch();
+    return result;
+  }, [rawAddHeart, refetch]);
+
+  const removeHeart = useCallback(async () => {
+    const result = await rawRemoveHeart({
+      variables: {
+        electionName: "reviewVoting2022",
+      }
+    });
+    const newHearts = result.data?.RemoveGivingSeasonHeart;
+    if (Array.isArray(newHearts)) {
+      setHearts(newHearts);
+    }
+    await refetch();
+  }, [rawRemoveHeart, refetch]);
+
+  const canAddHeart = !!currentUser && !isAddingHeart;
+  const [hoverPos, setHoverPos] = useState<{x: number, y: number} | null>(null);
+
+  const onMouseMove = useCallback(({target, clientX, clientY}: MouseEvent) => {
+    if (isValidTarget(target)) {
+      setHoverPos(normalizeCoords(clientX, clientY));
+    } else {
+      setHoverPos(null);
+    }
+  }, [normalizeCoords]);
+
+  const onMouseOut = useCallback(() => {
+    setHoverPos(null);
+  }, []);
+
+  const { totalCount } = useMulti({
+    terms: {
+      view: "reviewVotesFromUser",
+      userId: currentUser?._id,
+      year: REVIEW_YEAR.toString()
+    },
+    collectionName: "ReviewVotes",
+    fragmentName: 'reviewVoteFragment',
+    enableTotal: true,
+    skip: !currentUser,
+    limit: TARGET_REVIEW_NUM
+  });
+
+  const {flash} = useMessages();
+  const userHasVotedEnough = (totalCount || 0) >= TARGET_REVIEW_NUM;
+
+  const onClick = useCallback(async ({target, clientX, clientY}: MouseEvent) => {
+    if (isValidTarget(target)) {
+      const coords = normalizeCoords(clientX, clientY);
+      if (coords) {
+        if (!userHasVotedEnough) {
+          return flash("You need to have voted on at least six posts to place your ballot icon.")
+        }
+        const theta = Math.round((Math.random() * MAX_THETA * 2) - MAX_THETA);
+        const result = await addHeart(coords.x, coords.y, theta);
+        const newHearts = result.data?.AddGivingSeasonHeart;
+        if (Array.isArray(newHearts)) {
+          setHearts(newHearts);
+        }
+        setHoverPos(null);
+      }
+    }
+  }, [normalizeCoords, addHeart, flash, userHasVotedEnough]);
+
+  
+
+  return (
+    <AnalyticsContext pageSectionContext="header" siteEvent="givingSeason2023">
```

Comment:
I think leaving most of these variables as "givingSeason" instead of renaming is fine for a temporary feature, but you might want to update the `siteEvent` here in the analytics?

Reply:
Good catch, thanks :)